### ISSUE TITLE
exercise 1.2: inverted index in MapReduce

### ISSUE THREAD

**Issue by eubinean** [eubinecto/document-indexing#2]
## what?

>  writing a MapReduce job that creates an inverted index template program. 
- [x] Design the map and reduce operations for building an inverted index
  - First think in terms of pseudo code.
  - various tasks are analogous to `WordCount`
  - have a look at the codebase. You'll find some of them useful.
- [x] Implement the map reduce job in java
- [ ] Make it possible to do search on the inverted index.
- [ ] Review your implementations. what are the limitations? (will need to address these in the coursework). (have a look at computational complexity?)

On the basis of this inverted index, what you ultimately want to do is ranking the 7 Simpson Episodes (7 documents) with respect to a given query (short description of the episode). 

## why?
> context: Your boss is keen on the Simpsons. However, it can be difficult to remember the names of specific episodes. -> Have been asked to build an inverted index from Wikipedia pages of each episode, that will allow searches to be made for particular episodes using a search tool your colleague is building.
- Basically, for faster search on wikipedia page.

## how?
- [x] First, have a look at the data. untar two of them and get familiar with what data we want to break down. 
- [ ] what is **inverted index**? Study the what, why, and **how**(how first) of inverted index.

## Questions
- Why do we have to sort the terms? and why postings as secondary key?
- How do I search on inverted index?

**Comment by eubinean** [comment]
## The data
We have 6 `txt` files.  What contents do they contain?
```
Bart_the_Fink.txt.gz
Bart_the_General.txt.gz
Bart_the_Genius.txt.gz
Bart_the_Lover.txt.gz
Bart_the_Mother.txt.gz
Bart_the_Murderer.txt.gz
```
The content of `Bart_the_Fink.txt.gz` is [the wikipedia page of Bart The Fink](https://en.wikipedia.org/wiki/Bart_the_Fink) in pure text:

> Bart the Fink
From Wikipedia, the free encyclopedia Jump to: navigation, search "Bart the Fink" The Simpsons episode Episode?no. 143 Prod.?code 3F12 Orig.?airdate February 11, 1996 Showrunner(s) Bill Oakley Josh Weinstein Written?by Teleplay: John Swartzwelder Story: Bob Kushell[1] Directed?by Jim Reardon[2] Couch gag The couch is a fax machine that spews out a piece of paper with The Simpson family in a sitting position. Guest?star(s) Bob Newhart as himself Phil Hartman as Troy McClure DVD commentary Matt Groening Bill Oakley Josh Weinstein Jim Reardon David Silverman David X. Cohen Season 7 September 17, 1995?? May 19, 1996 "Who Shot Mr. Burns? (Part Two)" "Radioactive Man" "Home Sweet Homediddly- Dum-Doodily" "Bart Sells His Soul" "Lisa the Vegetarian" "Treehouse of Horror VI" "King-Size Homer" "Mother Simpson" "Sideshow Bob's Last Gleaming" "The Simpsons 138th Episode Spectacular" "Marge Be Not Proud" "Team Homer" "Two Bad Neighbors" "Scenes from the Class Struggle in Springfield" "Bart the Fink" "Lisa the Iconoclast" "Homer the Smithers" "The Day the Violence Died" "A Fish Called Selma" "Bart on the Road" "22 Short Films About Springfield" "Raging Abe Simpson and His Grumbling Grandson in 'The Curse of the Flying Hellfish'" "Much Apu About Nothing" "Homerpalooza" "Summer of 4 Ft. 2" List of all The Simpsons episodes Seasons 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 "Bart the Fink" is the fifteenth episode of The Simpsons' seventh season. It originally aired on the Fox network in the United States on February 11, 1996. In this episode, Bart ruins Krusty the Clown's career by accidentally exposing Krusty as one of ...



So, each text file contains Wikipedia entry of a Simpson episode (The Fink, The Mother, ... etc), each of which contains information about the episode(The story, Production, etc).

Okay, now I get a better understanding of the goal. What I have to do is **ranking** the episodes given a short description of an episode.

**Comment by eubinean** [comment]
# Inverted index
## what?

## how?


## why?
